src/main/java/edu/harvard/iq/dataverse/engine/command/impl/AbstractDatasetCommand.java

```diff
@@ -212,4 +221,28 @@ public void setDataset(Dataset dataset) {
     protected Timestamp getTimestamp() {
         return timestamp;
     }
+
+    protected void checkSystemMetadataKeyIfNeeded(DatasetVersion newVersion, DatasetVersion persistedVersion) throws IllegalCommandException {
+        Set<MetadataBlock> changedMDBs = DatasetVersionDifference.getBlocksWithChanges(newVersion, persistedVersion);
+        changedMDBs.forEach(mdb -> {
+            logger.fine(mdb.getName() + " has been changed");
+        });
+        
+        String smdbString = JvmSettings.METADATA_BLOCK_SYSTEM_METADATA_KEYS.lookupOptional().orElse(null);
+        if (smdbString != null) {
+            JsonObject systemMetadataBlocks = JsonUtil.getJsonObject(smdbString);
+            HttpServletRequest httpServletRequest = getRequest().getHttpServletRequest();
+            if (httpServletRequest != null) {
+                String mdKey = httpServletRequest.getParameter(MetadataBlockServiceBean.SYSTEM_MD_KEY);
```

Comment:
Wouldn't it make sense to move the parameter extraction to `DataverseRequest`, adding a method there and avoid a bit more if/else stuff by returning an `Optional<>`?